crates/lakekeeper/src/service/tasks/mod.rs

```diff
@@ -208,10 +273,11 @@ pub struct TaskInput {
 /// Metadata stored for each task in the database backend.
 /// This is separate from the task payload, which is specific to each task type.
 pub struct TaskMetadata {
-    pub warehouse_id: WarehouseId,
+    pub warehouse_id: Option<WarehouseId>,
+    pub project_id: ProjectId,
     pub parent_task_id: Option<TaskId>,
     pub entity_id: EntityId,
-    pub entity_name: Vec<String>,
+    pub entity_name: Option<Vec<String>>,
```

Comment:
As all entities in Lakekeeper are named, I don't think this should be optional. Think about the user: We return this as part of a list endpoint. Without an entity_name, the user just sees IDs. This is unfriendly and requires the user or the UI to run many more queries just to refresh all the machine readable IDs.